### PR TITLE
Add r-intrinsic (with dep r-salso)

### PR DESCRIPTION
**Pull request by leej3**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [ ] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by leej3** [comment]
I am not certain about static libraries in the build of the salso dynamic library (6.4MB). Rust and the cargo package manager are required during the build and lots of rust depenencies are used:

```
   Compiling autocfg v1.0.1
   Compiling cfg-if v1.0.0
   Compiling libc v0.2.108                ] 0/58: autocfg, cfg-if                                                                                                                       
   Compiling num-traits v0.2.14           ] 3/58: libc, autocfg                                                                                                                         
   Compiling crossbeam-utils v0.8.5       ] 6/58: libc, num-traits                                                                                                                      
   Compiling lazy_static v1.4.0           ] 8/58: num-traits, crossbeam-utils(build)                                                                                                    
   Compiling getrandom v0.2.3            ] 10/58: num-traits, crossbeam-utils                                                                                                           
   Compiling memoffset v0.6.4            ] 11/58: crossbeam-utils, getrandom                                                                                                            
   Compiling num-integer v0.1.44         ] 12/58: memoffset(build.rs), crossbeam-utils                                                                                                  
   Compiling proc-macro2 v1.0.32         ] 13/58: num-integer(build.rs), crossbeam-utils                                                                                                
   Compiling rand_core v0.6.3            ] 16/58: memoffset(build), proc-macro2(build.rs)                                                                                               
   Compiling unicode-xid v0.2.2          ] 18/58: proc-macro2(build), rand_core                                                                                                         
   Compiling crossbeam-epoch v0.9.5      ] 19/58: rand_core, unicode-xid                                                                                                                
   Compiling scopeguard v1.1.0           ] 24/58: proc-macro2, num-integer                                                                                                              
   Compiling ppv-lite86 v0.2.15          ] 25/58: scopeguard, num-integer                                                                                                               
   Compiling syn v1.0.81=>               ] 26/58: ppv-lite86, num-integer                                                                                                               
   Compiling crossbeam-queue v0.3.2      ] 27/58: ppv-lite86, syn(build.rs)                                                                                                             
   Compiling rand_chacha v0.3.1          ] 28/58: ppv-lite86, crossbeam-queue(build.rs)                                                                                                 
   Compiling quote v1.0.10=>             ] 32/58: crossbeam-epoch, rand_chacha                                                                                                          
   Compiling num-bigint v0.4.3           ] 33/58: crossbeam-epoch, quote                                                                                                                
   Compiling rawpointer v0.2.1           ] 34/58: quote, num-bigint(build.rs)                                                                                                           
   Compiling matrixmultiply v0.3.2       ] 35/58: rawpointer, num-bigint(build.rs)                                                                                                      
   Compiling crossbeam-deque v0.8.1      ] 38/58: syn, matrixmultiply                                                                                                                   
   Compiling rand v0.8.4=======>         ] 39/58: crossbeam-deque, syn                                                                                                                  
   Compiling crossbeam-channel v0.5.1    ] 41/58: syn, crossbeam-queue                                                                                                                  
   Compiling num-complex v0.4.0=>        ] 42/58: syn, crossbeam-channel                                                                                                                
   Compiling ndarray v0.15.3=====>       ] 43/58: syn, num-complex                                                                                                                      
   Compiling roxido_macro v0.4.2 (/home/conda/staged-recipes/build_artifacts/r-salso_1675934485960/work/src/rustlib/roxido_macro)                                                       
   Compiling crossbeam v0.8.1====>       ] 45/58: roxido_macro, ndarray                                                                                                                 
   Compiling dahl-partition v0.5.3>      ] 46/58: ndarray, crossbeam                                                                                                                    
   Compiling rand_pcg v0.3.1=======>     ] 48/58: num-bigint, ndarray                                                                                                                   
   Compiling approx v0.5.0=========>     ] 49/58: num-bigint, rand_pcg                                                                                                                  
   Compiling approx v0.4.0==========>    ] 50/58: num-bigint, approx                                                                                                                    
   Compiling num_cpus v1.13.0=======>    ] 51/58: num-bigint, approx                                                                                                                    
   Compiling dahl-bellnumber v0.1.1==>   ] 52/58: num-bigint, num_cpus                                                                                                                  
   Compiling dahl-salso v0.6.6=======>   ] 53/58: num-bigint, dahl-bellnumber                                                                                                           
   Compiling roxido v0.4.2 (/home/conda/staged-recipes/build_artifacts/r-salso_1675934485960/work/src/rustlib/roxido)                                                                   
   Compiling rustlib v0.1.0 (/home/conda/staged-recipes/build_artifacts/r-salso_1675934485960/work/src/rustlib)                                                                         
    Finished release [optimized] target(s) in 55.35sstlib                                                                                                                               
```

The so compilation seems to use rust deps for linking:

```
x86_64-conda-linux-gnu-cc -shared -L$PREFIX/lib/R/lib -Wl,-O2 -Wl,--sort-common -Wl,--as-needed -Wl,-z,relro -Wl,-z,now -Wl,--disable-new-dtags -Wl,--gc-sections -Wl,--allow-shlib-undefined -Wl,-rpath,$PREFIX/lib -Wl,-rpath-link,$PREFIX/lib -L$PREFIX/lib -o salso.so shim.o -Lrustlib/target/release -lrustlib -L$PREFIX/lib/R/lib -lR
installing to /home/conda/staged-recipes/build_artifacts/r-salso_1675934485960/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pla/lib/R/library/00LOCK-work/00new/salso/libs
```

I see no requirement for anything to do with rust at run time though

```
$ ldd ./lib/R/library/salso/libs/salso.so
        linux-vdso.so.1 (0x00007fff37762000)
        libR.so => [...]/envs/intrinsic/./lib/R/library/salso/libs/../../../lib/libR.so (0x00007f0c98735000)
        libgcc_s.so.1 => [...]/envs/intrinsic/./lib/R/library/salso/libs/../../../../libgcc_s.so.1 (0x00007f0c9871c000)
        libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f0c9852d000)
        /lib64/ld-linux-x86-64.so.2 (0x00007f0c98e2e000)
        libblas.so.3 => [...]/envs/intrinsic/./lib/R/library/salso/libs/../../../lib/../../libblas.so.3 (0x00007f0c96285000)
        libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007f0c9613f000)
        libreadline.so.8 => [...]/envs/intrinsic/./lib/R/library/salso/libs/../../../lib/../../libreadline.so.8 (0x00007f0c960e8000)
        libpcre2-8.so.0 => [...]/envs/intrinsic/./lib/R/library/salso/libs/../../../lib/../../libpcre2-8.so.0 (0x00007f0c96048000)
        liblzma.so.5 => [...]/envs/intrinsic/./lib/R/library/salso/libs/../../../lib/../../liblzma.so.5 (0x00007f0c9601f000)
        libbz2.so.1.0 => [...]/envs/intrinsic/./lib/R/library/salso/libs/../../../lib/../../libbz2.so.1.0 (0x00007f0c9600b000)
        libz.so.1 => [...]/envs/intrinsic/./lib/R/library/salso/libs/../../../lib/../../libz.so.1 (0x00007f0c95fef000)
        librt.so.1 => /lib/x86_64-linux-gnu/librt.so.1 (0x00007f0c95fe5000)
        libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007f0c95fdf000)
        libiconv.so.2 => [...]/envs/intrinsic/./lib/R/library/salso/libs/../../../lib/../../libiconv.so.2 (0x00007f0c95ef8000)
        libicuuc.so.70 => [...]/envs/intrinsic/./lib/R/library/salso/libs/../../../lib/../../libicuuc.so.70 (0x00007f0c95cf6000)
        libicui18n.so.70 => [...]/envs/intrinsic/./lib/R/library/salso/libs/../../../lib/../../libicui18n.so.70 (0x00007f0c959cc000)
        libgomp.so.1 => [...]/envs/intrinsic/./lib/R/library/salso/libs/../../../lib/../../libgomp.so.1 (0x00007f0c95991000)
        libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007f0c9596f000)
        libgfortran.so.5 => [...]/envs/intrinsic/./lib/R/library/salso/libs/../../../lib/../.././libgfortran.so.5 (0x00007f0c957c4000)
        libtinfo.so.6 => [...]/envs/intrinsic/./lib/R/library/salso/libs/../../../lib/../.././libtinfo.so.6 (0x00007f0c95784000)
        libicudata.so.70 => [...]/envs/intrinsic/./lib/R/library/salso/libs/../../../lib/../.././libicudata.so.70 (0x00007f0c93b65000)
        libstdc++.so.6 => [...]/envs/intrinsic/./lib/R/library/salso/libs/../../../lib/../.././libstdc++.so.6 (0x00007f0c939b1000)
        libquadmath.so.0 => [...]/envs/intrinsic/./lib/R/library/salso/libs/../../../lib/../../././libquadmath.so.0 (0x00007f0c93977000)
```

I have tested the installed packages and they work fine without rust in the environment though which makes me think that there is some static linking going on. Not sure how to troubleshoot this further.

**Comment by leej3** [comment]
@conda-forge/help-r

File: recipes/r-intrinsic/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+build:
+  merge_build_host: True  # [win]
+  number: 0
+  rpaths:
+    - lib/R/lib/
+    - lib/
+
+requirements:
+  build:
+    - _openmp_mutex                # [linux]
```
Comment (dbast) [inline recipes/r-intrinsic/meta.yaml:24]: see https://conda-forge.org/docs/maintainer/knowledge_base.html#openmp
```suggestion
    - llvm-openmp  # [osx]
    - libgomp      # [linux]
```

File: recipes/r-intrinsic/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    - r-ggplot2
+    - r-knitr
+    - r-latex2exp
+    - r-reshape2
+    - r-rlang
+    - r-salso
+  run:
+    - r-base
+    - {{ native }}gcc-libs         # [win]
+    - llvm-openmp                  # [osx]
```
Comment (dbast) [inline recipes/r-intrinsic/meta.yaml:51]:
```suggestion
```

**Comment by dbast** [comment]
2 recipes, so builds require a retrigger after merge in the feedstocks

**Pull request merged**